docs/contributor/arch/025-minimal-privileges-for-log-agent.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Minimal Privileges for Log Agent"
+date: 2025-01-15
+---
+
+
+# 25. Minimal Privileges for Log Agent
+
+## Context
+
+The log-agent is a component that reads logs directly from the node filesystem and forwards it to a configured sink. It requires specific permissions to access the logs of pods in the cluster. The goal is to define the minimal set of privileges required for the Log Reader to function correctly without granting excessive permissions.
+
+## Current Situation
+
+The log-agent currently runs with elevated privileges, which allows it to access all pod logs in the cluster. This is not ideal from a security perspective.
+The elevated privileges are necessary to ensure that the log-agent can:
+
+1. Maintain a file cache which is persistent across restarts, allowing it to resume reading logs from where it left off.
+2. Read logs from all pods, including those that are not in the same namespace as the log-agent.
+
+## Possible Solutions
+
+### Cache Directory Permissions
+The log-agent needs to be able to write to a filesystem location that is accessible to it. As our log-agent runs as a daemonset, it has access to the host filesystem. 
+
+### Option a
+The log-agent can uses a directory in `/var/lib/` for its file cache, which is NOT writeable by the daemonset user by default. Granting write access to this directory can be done using an init container, that runs chmod on the directory before the log-agent starts. To allow changing the premissions the init container needs to run with elevated privileges, which is acceptable as it only runs once during the pod startup.
+
+### Option b
+Another option it to store those cache entries in a subdirectory of `/var/tmp/` which is writable by the daemonset user by default. This would not require an init container. To circumvent the issue of the log-agent not being able to write to the folder the directory cannot be created by the kubelet at startup, but must be created by the log-agent itself. This can be done using the `create_directory` option of the `file_storage` extension.
+
+telemetry-log-agent change:
+```yaml
+...
+    extensions:
+        file_storage:
+            create_directory: true
+            directory: /var/lib/telemetry-log-agent/file-log-receiver
+...
+```
+
+### Log Reading Permissions
+
+The log-agent needs to read logs from all pods in the cluster. Those logs are stored in the `/var/log/pods` on the node filesystem. The log-agent needs to have read access to this directory. The default permissions of this directory allow only acces to the root user and root group. To allow the log-agent to read the logs, we can either:
+
+### Option a
+Grant read access to the `/var/log/pods` directory to the log-agent user. This can be done by running the log-agent with the `runAsGroup: 0` security context option in the pod spec.
```

Comment:
```suggestion
Grant read access to the `/var/log/pods` directory to the log agent user. This can be done by running the log agent with the `runAsGroup: 0` security context option in the Pod spec.
```